sys/sched.c

```diff
@@ -59,8 +59,11 @@ void sched_switch(thread_t *newtd) {
 
   if (newtd == NULL) {
     newtd = runq_choose(&runq);
-    if (newtd)
+    if (newtd) {
+      if(newtd->td_type == TD_USER && newtd->user_map)
+          set_active_vm_map(newtd->user_map);
```

Comment:
Switching address space should be done in [ctx_switch] (https://github.com/cahirwpz/mimiker/blob/master/mips/switch.S#L37). For reference look into FreeBSD version of [ctx_switch](http://bxr.su/FreeBSD/sys/mips/mips/swtch.S#354).